### PR TITLE
Sync with upstream v262

### PR DESCRIPTION
**Pull request by Frzk**
Related to #54
Deployed here: https://fku-dummy-ruby.osc-fr1.scalingo.io/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,17 @@
...
+## v262 (2023/11/08)
+
+- Warn when relying on default Node.js or Yarn versions (https://github.com/heroku/heroku-buildpack-ruby/pull/1401)
+- Warn when default Node.js or Yarn versions change (https://github.com/heroku/heroku-buildpack-ruby/pull/1401)
+
+## v261 (2023/11/02)
+
+- JRuby 9.4.5.0 is now available
+- JRuby 9.3.13.0 is now available
+- Default Node.js version now 20.9.0
```
Comment (EtienneM) [inline CHANGELOG.md:14]: AH :smile:

File: lib/language_pack/ruby.rb [hunk 2]
```diff
@@ -1019,6 +1019,26 @@ def add_node_js_binary
...
+      end
+
+      warn(<<~WARNING, inline: true)
+        Installing a default version (#{version}) of Node.js.
+        This version is not pinned and can change over time, causing unexpected failures.
+
+        Heroku recommends placing the `heroku/nodejs` buildpack in front of
+        `heroku/ruby` to install a specific version of node:
+
+        https://devcenter.heroku.com/articles/ruby-support#node-js-support
```
Comment (EtienneM) [inline lib/language_pack/ruby.rb:1039]: issue: I would update this message to better match Scalingo

File: lib/language_pack/ruby.rb [hunk 3]
```diff
@@ -1029,6 +1049,27 @@ def add_yarn_binary
...
+
+      warn(<<~WARNING, inline: true)
+        Installing a default version (#{version}) of Yarn
+        This version is not pinned and can change over time, causing unexpected failures.
+
+        Heroku recommends placing the `heroku/nodejs` buildpack in front of the `heroku/ruby`
+        buildpack as it offers more comprehensive Node.js support, including the ability to
+        customise the Node.js version:
+
+        https://devcenter.heroku.com/articles/ruby-support#node-js-support
```
Comment (EtienneM) [inline lib/language_pack/ruby.rb:1070]: issue: I would update this message to better match Scalingo

File: lib/language_pack/ruby.rb (hunk 2 above)
Comment (Frzk) [reply lib/language_pack/ruby.rb]: Right!

File: lib/language_pack/ruby.rb (hunk 3 above)
Comment (Frzk) [reply lib/language_pack/ruby.rb]: Also right!

File: CHANGELOG.md (hunk 1 above)
Comment (yohann-bacha) [reply CHANGELOG.md:14]: AH.

**Pull request merged**